app/models/piece.rb

```diff
@@ -12,5 +12,56 @@ class Piece < ActiveRecord::Base
 	def self.types
 		%w(Pawn Knight Bishop Rook Queen King)
 	end
+
+	# Determine if there are obstructions "between" two chess pieces
+	# to validate a piece can succesfully be moved.  Accepts x and y coordinates 
+	# for both pieces being compared.  Returns false if no obstruction exists, true if obstruction exists.
+	def is_obstructed?(start_point, end_point)
```

Comment:
Correct me if I'm wrong, but I don't think this method accounts for obstructions along a path.  We aren't just concerned with the two endpoints but rather any obstruction including the endpoint along a path.  True that a piece can never move through an obstruction but that doesn't mean a user won't try, so you need this to account for that so you can tell the user the location they have chosen is invalid.